### DISCUSSION TITLE
Recommended way to know if a Thread device is connected

### DISCUSSION THREAD

**Discussion by OlivierGre** [openthread/openthread#8324] [Q&A] [answered]
What is the recommended way to know if a Thread device is connected to a thread network?

In [this discussion](https://github.com/openthread/openthread/issues/6912#issuecomment-896485314), there is this code:

```
static bool isOtBrConnected(void) {
  otBorderRouterConfig borderRouterConfig;
  otNetworkDataIterator iter = OT_NETWORK_DATA_ITERATOR_INIT;
  if (otNetDataGetNextOnMeshPrefix(thread_ot_instance_get(), &iter, &borderRouterConfig) == OT_ERROR_NONE)
    return true;
  else
    return false;
}
```

Or is there an OT API for that?
Thanks

**Comment by jwhui** [comment, marked as answer]
It really depends on how you define "connected". For example, it could be:

1. Attached to a Thread network with one or more neighbors
2. Attached to a Thread network with a Thread Border Router providing external connectivity
3. Attached to a Thread network with a Thread Border Router providing public Internet connectivity

For (1), simply checking the neighbor table is at least one entry should be sufficient.
For (2), checking the Thread Network Data for an on-mesh prefix and an external route should be sufficient.
For (3), checking the Thread Network Data for an on-mesh prefix and a default route should be sufficient.

Beyond that, others define "connected" as being able to establish a session with some other endpoint, and that endpoint may be on the same Thread network, same local IP network, or on the public Internet.

**Comment by OlivierGre** [comment]
Ok thank you Jonathan. You're right that it was important to clarify the meaning of "connected". In my case I wanted to know if the device is connected to a Thread Border Router so this is the case 2. For my demos, it is not important if we have internet connectivity or not.
